sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -902,7 +902,7 @@ object IntervalUtils extends SparkIntervalUtils {
       case DAY => Decimal(v / MICROS_PER_DAY)
       case HOUR => Decimal(v / MICROS_PER_HOUR)
       case MINUTE => Decimal(v / MICROS_PER_MINUTE)
-      case SECOND => Decimal(v, Decimal.MAX_LONG_DIGITS, 6)
+      case SECOND => Decimal(v, Decimal.MAX_LONG_DIGITS + 1, 6)
```

Comment:
How does this change fix the problem?

Reply:
This function is called [when casting from DT Interval to decimal](https://github.com/apache/spark/blob/0c18fc072b05671bc9c74a43de49b563a1ef7907/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala#L993-L998).

The logic is that the interval is first cast to a decimal of arbitrary precision and scale and then these arbitrary parameters are changed to the target parameters. However, currently, in the specific case where end-field is SECOND, and the interval happens to be large (19 digits, which is the worst case), it doesn't fit into a decimal of precision `Decimal.MAX_LONG_DIGITS` which is 18.

Reply:
This only affects explicit type casting right? Will it change the behavior of schema inference and thus change the query result schema?

Reply:
The only place where this function (dayTimeIntervalToDecimal) is called is in `Cast.scala` when casting from dt-intervals to decimal. Even there, the [precision of the result is changed](https://github.com/apache/spark/blob/0c18fc072b05671bc9c74a43de49b563a1ef7907/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala#L995-L998) from the intermediate precision (which is what I am changing here) to a target precision (which my code does not touch).

There is no reason to believe that my code changes anything about the output.